### DISCUSSION TITLE
JSON items

### DISCUSSION THREAD

**Discussion by Kolhun** [nlohmann/json#4121] [Q&A]
Hello everyone, tell me there is a string, I translate it into JSON (json::parse). The question is that the keys I need are located in the "data" object how do I access the keys that are located in it? And work directly with it
![262970446-5b5dab81-4e34-4460-aac8-6fe892d305be](https://github.com/nlohmann/json/assets/36815965/4555113c-1c6b-4849-bda4-46704a57d047)
![262970394-69b0dad9-8eea-4886-b9bf-13fde32cc624](https://github.com/nlohmann/json/assets/36815965/0132b28a-b7cc-4f28-b672-a65496e83791)

**Comment by nlohmann** [comment]
After json j = json::parse(str), access the nested object with j["data"] (or j.at("data") if you want a bounds-checked access that throws on a missing key). To work with all the keys inside it, iterate with for (auto& [key, value] : j["data"].items()) { ... } (structured bindings, C++17) or the equivalent it->key()/it->value() form with explicit iterators. See docs/mkdocs/docs/features/iterators.md for the full iteration API.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
